manifests/config.pp

```diff
@@ -52,13 +56,25 @@
       "${git_home}/gitlab/tmp",
       "${git_home}/gitlab/tmp/pids",
       "${git_home}/gitlab/tmp/sockets",
-      "${git_home}/gitlab/public",
-      "${git_home}/gitlab/public/uploads",
+      "${git_home}/gitlab/public",      
     ]:
     ensure => directory,
     mode   => '0755',
   }
-
+  
+  #gitlab does not provide an option to configure the uploads directory location, so create a symlink to
+  #the desired folder if specified (otherwise, simply ensure the default uploads folder is there)
+  $gitlab_uploads_path_type = $gitlab_uploads_folder ? {
+    undef   => 'directory',
+    default => 'link',
+  }
+  file { "${git_home}/gitlab/public/uploads":
+    ensure => $gitlab_uploads_path_type,
+    target => $gitlab_uploads_folder,
+    mode   => '0755',
```

Comment:
I think that this is meant to be `0750`. See https://github.com/gitlabhq/gitlabhq/blob/3f05c22f5612c26ff3707aa16eb0f3c8be18ec93/lib/tasks/gitlab/check.rake#L297